tests/requests_tests/tests.py

```diff
@@ -694,18 +710,30 @@ def read(self, size=-1, /):
             request.body
 
     def test_set_encoding_clears_POST(self):
-        payload = FakePayload("name=Hello Günter")
+        payload = FakePayload(
+            "\r\n".join(
+                [
+                    f"--{BOUNDARY}",
+                    'Content-Disposition: form-data; name="name"',
+                    "",
+                    "Hello Günter",
+                    f"--{BOUNDARY}--",
+                    "",
+                ]
+            )
+        )
         request = WSGIRequest(
             {
                 "REQUEST_METHOD": "POST",
-                "CONTENT_TYPE": "application/x-www-form-urlencoded",
+                "CONTENT_TYPE": MULTIPART_CONTENT,
                 "CONTENT_LENGTH": len(payload),
                 "wsgi.input": payload,
             }
         )
         self.assertEqual(request.POST, {"name": ["Hello Günter"]})
         request.encoding = "iso-8859-16"
-        self.assertEqual(request.POST, {"name": ["Hello GĂŒnter"]})
+        # Check out #14035.
```

Comment:
Could we expand this comment a little bit? Specifically, when reading the corresponding ticket, it seems that this may go against [the docs](https://docs.djangoproject.com/en/4.2/ref/request-response/#django.http.HttpRequest.encoding)?

> You can write to this attribute to change the encoding used when accessing the form data. Any subsequent attribute accesses (such as reading from [GET](https://docs.djangoproject.com/en/4.2/ref/request-response/#django.http.HttpRequest.GET) or [POST](https://docs.djangoproject.com/en/4.2/ref/request-response/#django.http.HttpRequest.POST)) will use the new encoding value

Reply:
> Specifically, when reading the corresponding ticket, it seems that this may go against the docs?

Yes, it is. This is why ticket-14035 is a bug waiting for being fixed. TBH, I'm not sure it can be fixed.
I tried to make it clearer.